domain/machine/state/state_test.go

```diff
@@ -776,8 +776,48 @@ func (s *stateSuite) TestSetLXDProfiles(c *gc.C) {
 	err = s.state.SetLXDProfiles(context.Background(), "deadbeef", []string{"profile1", "profile2"})
 	c.Assert(err, jc.ErrorIsNil)
 
-	profiles, err := s.state.LXDProfiles(context.Background(), "deadbeef")
+	// Check that the profile names are in the machine_lxd_profile table.
+	db := s.DB()
+	rows, err := db.Query("SELECT name FROM machine_lxd_profile WHERE machine_uuid = 'deadbeef'")
+	defer rows.Close()
+	c.Assert(err, jc.ErrorIsNil)
+	var profiles []string
+	for rows.Next() {
+		var profile string
+		err := rows.Scan(&profile)
+		c.Assert(err, jc.ErrorIsNil)
+		profiles = append(profiles, profile)
+	}
+	c.Check(profiles, gc.HasLen, 2)
+	c.Check(profiles, jc.SameContents, []string{"profile1", "profile2"})
+}
+
+func (s *stateSuite) TestSetLXDProfilesPartial(c *gc.C) {
+	err := s.state.CreateMachine(context.Background(), "666", "", "deadbeef")
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Insert a single lxd profile.
+	db := s.DB()
+	_, err = db.Exec(`INSERT INTO machine_lxd_profile VALUES
+("deadbeef", "profile1")`)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.SetLXDProfiles(context.Background(), "deadbeef", []string{"profile1", "profile2"})
+	// This shouldn't fail, but add the missing profile to the table.
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Check that the profile names are in the machine_lxd_profile table.
+	rows, err := db.Query("SELECT name FROM machine_lxd_profile WHERE machine_uuid = 'deadbeef'")
+	defer rows.Close()
 	c.Assert(err, jc.ErrorIsNil)
+	var profiles []string
+	for rows.Next() {
+		var profile string
+		err := rows.Scan(&profile)
+		c.Assert(err, jc.ErrorIsNil)
+		profiles = append(profiles, profile)
+	}
+	c.Check(profiles, gc.HasLen, 2)
```

Comment:
**todo:** remove hasLen check, I think it is redundant with the SameContent check (which already do it)

Reply:
Fixed in 82a3a302d8.